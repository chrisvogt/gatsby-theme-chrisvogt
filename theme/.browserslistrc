# Modern browsers that support spread operators and classes natively
# This reduces the need for @babel/plugin-transform-spread and @babel/plugin-transform-classes

# Chrome/Edge (last 2 versions)
last 2 Chrome versions
last 2 Edge versions

# Firefox (last 2 versions)
last 2 Firefox versions

# Safari (last 2 versions)
last 2 Safari versions

# iOS Safari (last 2 versions)
last 2 iOS versions

# Android Chrome (last 2 versions)
last 2 Android versions

# Don't support IE
not IE 11
not IE 10
not IE 9

# Don't support very old versions
> 0.5%
not dead 