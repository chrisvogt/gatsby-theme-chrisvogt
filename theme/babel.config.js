// This is the Babel configuration, which is used by Jest to transpile the theme
// source code when running tests. This is not utilized by GatsbyJS. To use your
// own Babel config with GatsbyJS, please review the following documentation:
// https://www.gatsbyjs.com/docs/babel/
module.exports = {
  presets: ['@babel/preset-env', '@babel/preset-react']
}
