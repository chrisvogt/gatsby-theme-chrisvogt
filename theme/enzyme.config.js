const { configure, mount, render, shallow } = require('enzyme')
const Adapter = require('enzyme-adapter-react-16')

configure({
  adapter: new Adapter()
})

// loader shim
global.___loader = { enqueue: () => {} }

global.mount = mount
global.render = render
global.shallow = shallow

console.error = message => {
  throw new Error(message)
}
