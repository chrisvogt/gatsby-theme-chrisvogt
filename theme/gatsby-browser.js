import 'prismjs/themes/prism-solarizedlight.css'
import 'prismjs/plugins/line-numbers/prism-line-numbers.css'

import './src/styles/global.css'

export { default as wrapRootElement } from './wrapRootElement'

// See https://fossies.org/linux/gatsby/examples/using-reach-skip-nav/README.md
export const onRouteUpdate = ({ location, prevLocation }) => {
  if (prevLocation !== null) {
    const skipLink = document.querySelector('[data-reach-skip-link]') // Ccomes with the <SkipNavLink> component.
    if (skipLink) {
      skipLink.focus()
    }
  }
}
