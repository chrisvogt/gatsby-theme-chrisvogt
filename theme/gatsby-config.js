const path = require('path')

module.exports = () => ({
  siteMetadata: {
    avatarURL:
      'https://res.cloudinary.com/chrisvogt/image/upload/v1573025803/avatar_2x_srlojo.png',
    baseURL: 'https://www.privatesphere.io',
    description:
      'A GatsbyJS blog theme with built-in social widgets for Instagram, Goodreads, GitHub, and Spotify.',
    footerText: 'Made in San Francisco',
    headline: 'Private Sphere',
    imageURL: '',
    languageCode: 'en',
    social: {
      twitterUsername: ''
    },
    subhead: 'A Gatsby blog theme with built-in social widgets',
    title: 'Private Sphere',
    titleTemplate: '%s Â· Private Sphere',
    widgets: {
      github: {
        username: '',
        widgetDataSource: ''
      },
      goodreads: {
        username: '',
        widgetDataSource: ''
      },
      instagram: {
        username: '',
        widgetDataSource: ''
      },
      spotify: {
        username: '',
        widgetDataSource: ''
      },
      steam: {
        username: '',
        widgetDataSource: ''
      }
    }
  },
  plugins: [
    {
      resolve: 'gatsby-plugin-page-creator',
      options: {
        path: path.join(__dirname, 'src/pages')
      }
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        path: path.join(__dirname, 'src/data')
      }
    },
    {
      resolve: 'gatsby-source-filesystem',
      options: {
        path: `content`,
        name: `content`
      }
    },
    `gatsby-plugin-sharp`,
    `gatsby-transformer-sharp`,
    {
      resolve: 'gatsby-plugin-mdx',
      options: {
        gatsbyRemarkPlugins: [
          `gatsby-remark-prismjs`,
          `gatsby-remark-images`,
          `gatsby-remark-copy-linked-files`
        ],
        plugins: [
          // FIX(cvogt): this plugin is defined here as a temporary fix for the bug
          // described in gatsbyjs/gatsby#15486
          `gatsby-remark-images`
        ]
      }
    },
    'gatsby-plugin-emotion',
    'gatsby-plugin-react-helmet',
    'gatsby-theme-style-guide',
    'gatsby-transformer-json',
    `gatsby-plugin-theme-ui`
  ]
})
