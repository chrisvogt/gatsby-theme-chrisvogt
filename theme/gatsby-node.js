const path = require('path')
const startCase = require('lodash/startCase')

const getNodePath = node => {
  let nodePath = '/'

  const category = node.fields.category
  if (category) {
    nodePath += `${category}/`
  }

  const slug = node.frontmatter.slug
  if (slug) {
    nodePath += slug
  }

  return nodePath
}

exports.createPages = async ({ actions, graphql, reporter }) => {
  actions.createPage({
    component: path.resolve('../theme/src/templates/home.js'),
    path: '/'
  })

  const result = await graphql(
    `
      {
        allMdx {
          edges {
            node {
              id
              fields {
                category
                slug
                type
              }
              frontmatter {
                slug
              }
              internal {
                contentFilePath
              }
            }
          }
        }
      }
    `
  )

  if (result.errors) {
    reporter.panic('error loading content', result.errors)
    return
  }

  result.data.allMdx.edges.forEach(({ node }) => {
    const nodePath = getNodePath(node)
    const template =
      node.fields.type === 'media'
        ? path.resolve('../theme/src/templates/media.js')
        : path.resolve('../theme/src/templates/post.js')

    actions.createPage({
      path: nodePath ? nodePath : '/',
      component: `${template}?__contentFilePath=${node.internal.contentFilePath}`,
      context: {
        id: node.id
      }
    })
  })
}

exports.onCreateNode = ({ node, getNode, actions, reporter }) => {
  const { createNodeField } = actions

  if (node.internal.type === `Mdx`) {
    const parent = getNode(node.parent)
    const title = node.frontmatter.title || startCase(parent.name)

    const slug = node.frontmatter.slug
    if (!slug) {
      reporter.panic(
        `Can not create node with title: ${title} there is no relative path or frontmatter to set the "slug" field`
      )
      return
    }

    if (slug === 'index') {
      slug = ''
    }

    const category = node.frontmatter.category
    if (category) {
      createNodeField({
        name: `category`,
        node,
        value: category
      })
    }

    const type = node.frontmatter.type
    if (type) {
      createNodeField({
        name: `type`,
        node,
        value: type
      })
    }

    createNodeField({
      name: `slug`,
      node,
      value: slug
    })

    createNodeField({
      name: 'id',
      node,
      value: node.id
    })

    if (node.internal.type === `Mdx`) {
      createNodeField({
        name: 'path',
        node,
        value: getNodePath(node)
      })
    }

    createNodeField({
      name: 'title',
      node,
      value: title
    })
  }
}
