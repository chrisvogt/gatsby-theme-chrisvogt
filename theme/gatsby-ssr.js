export { default as wrapRootElement } from './wrapRootElement'
