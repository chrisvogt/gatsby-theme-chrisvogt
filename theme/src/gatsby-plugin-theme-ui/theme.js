import { tailwind } from '@theme-ui/presets'
import { merge } from 'theme-ui'

const fonts = {
  sans: '-apple-system, BlinkMacSystemFont, avenir next, avenir, helvetica neue, helvetica, Ubuntu, roboto, noto, segoe ui, arial, sans-serif',
  serif: 'Crimson Text, Georgia, Iowan Old Style, Apple Garamond, Baskerville, Times New Roman, Droid Serif, Times, Source Serif Pro, serif, Apple Color Emoji, Segoe UI Emoji, Segoe UI Symbol',
  mono: 'Menlo, Consolas, Monaco, Liberation Mono, Lucida Console, monospace'
}

export const floatOnHover = {
  transition: `all 200ms ease-in-out`,

  '&:hover, &:focus': {
    transform: `scale(1.015)`,
    boxShadow: `lg`
  }
}

const GradientBanner = {
  /**
   * Gradient animation created by @bibby0912.
   * Visit https://codepen.io/bibby0912/pen/mErWyA
   */
  maxWidth: `100%`,
  height: `340px`,
  border: `20px solid transparent`,
  boxSizing: `border-box`,
  padding: `1rem`,
  display: `flex`,
  justifyContent: `center`,
  alignItems: `center`,
  color: `black`,

  backgroundImage: `
  linear-gradient(white, white),
  linear-gradient(180deg, cornflowerblue, purple 50%, cornflowerblue)`,

  backgroundRepeat: `no-repeat`,
  backgroundSize: `100% 100%, 100% 200%`,
  backgroundPosition: `0 0, 0 100%`,
  backgroundOrigin: `padding-box, border-box`,
  animation: `highlight 3s infinite alternate`,

  '@keyframes highlight': {
    '100%': {
      backgroundPosition: `0 0, 0 0`
    }
  }
}

const glassmorhismPanel = {
  borderRadius: '10px',
  backdropFilter: 'blur(10px)',
  WebkitBackdropFilter: 'blur(10px)',
  boxShadow: '0 4px 6px rgba(0, 0, 0, 0.1)',
  border: '1px solid rgba(255, 255, 255, 0.15)'
}

export const card = {
  borderRadius: 'card',
  background: `var(--theme-ui-colors-panel-background)`,
  color: 'var(--theme-ui-colors-panel-text)',
  boxShadow: `default`,
  flexGrow: 1,
  padding: 3,
  fontSize: [1, 2],
  textDecoration: `none`
}

export const metricCard = {
  backgroundColor: `var(--theme-ui-colors-panel-background)`,
  boxShadow: `none`,
  color: 'var(--theme-ui-colors-panel-text)',
  span: {
    fontFamily: `heading`,
    fontWeight: `bold`,
    padding: 2
  }
}

export const PostCard = {
  ...card,
  ...floatOnHover,
  ...glassmorhismPanel,
  backgroundColor: `var(--theme-ui-colors-panel-background)`,
  color: 'var(--theme-ui-colors-panel-text)',
  display: `flex`,
  height: `100%`,
  flexDirection: `column`,
  '.card-media': {
    mb: 2,
    height: `100%`,
    overflow: `hidden`
  },
  '.read-more-icon': {
    display: `inline`,
    transition: `all 250ms ease-in`,
    opacity: 0,
    paddingLeft: 0
  },
  '&:hover .read-more-icon': {
    opacity: 1,
    paddingLeft: `8px`
  }
}

export default merge(tailwind, {
  useColorSchemeMediaQuery: true,

  badges: {
    primary: {
      color: 'background',
      bg: 'primary'
    },
    outline: {
      color: 'primary',
      bg: 'transparent',
      boxShadow: 'inset 0 0 0 1px',
      fontSize: 1
    }
  },

  buttons: {
    primary: {
      color: 'background',
      bg: 'primary',
    },
    secondary: {
      color: 'background',
      bg: 'secondary',
    },
    gray: {
      color: 'background',
      bg: 'gray',
    },
  },

  cards: {
    /* The <Card /> default style. Used when no variant is defined. */
    primary: {
      ...card
    },

    actionCard: {
      ...card,
      ...floatOnHover,
      ...glassmorhismPanel,
      borderLeft: theme => `2px solid ${theme.colors.primary}`,
      a: {
        ':hover': 'pointer'
      }
    },

    metricCard: {
      ...card,
      ...metricCard
    },

    /* The following styles represent specific card components, indicated in PascalCase. */

    UserProfile: {
      padding: theme => [`none`, `0 ${theme.space[3]} 0 0`],
      border: `none`,
      color: `white`,
      background: `none`
    },

    UserProfileDark: {
      ...card,
      padding: theme => [`none`, `0 ${theme.space[3]} 0 0`],
      color: `white`,
      borderBottom: `none`,
      backgroundColor: `none`,
      boxShadow: `none`
    },

    StatusCardDark: {
      backgroundColor: `#1e2530`
    },

    PostCard
  },

  colors: {
    accent: `deeppink`,
    background: '#fcb8a2',
    'panel-background': `rgba(255, 229, 224, 0.10)`,
    'panel-divider': theme => `1px solid rgba(255, 229, 224, 0.17)`,
    'panel-highlight': theme => theme.colors.gray[1],
    modes: {
      dark: {
        background: `#1e1e2f`,
        'panel-background': `rgba(1, 1, 1, 0.28)`,
        'panel-divider': theme => `1px solid ${theme.colors.gray[8]}`,
        'panel-highlight': theme => theme.colors.gray[8],
        primary: `#1E90FF`,
        text: `#fff`,
        textMuted: '#d8d8d8'
      }
    },
    primary: `#422EA3`,
    secondary: `#711E9B`,
    secondaryGradient: `linear-gradient(45deg, #4527a0 0%, #711e9b 100%)`,
    tableText: `#111`,
    text: `#111`
  },

  fonts: {
    body: fonts.serif,
    heading: fonts.sans,
    monospace: fonts.mono,
    sans: fonts.sans,
    serif: fonts.serif
  },

  fontSizes: ['.875rem', '1rem', '1.25rem', '1.375rem', '1.5rem', '1.875rem', '2.25rem', '3rem', '4rem', '4.5rem'],

  layout: {
    container: {
      maxWidth: [``, `98%`, ``, ``, `1440px`],
      py: [2, 3],
      px: [3, 4]
    }
  },

  links: {
    homeNavigation: {
      color: `primary`,
      display: `block`,
      py: 2,
      textDecoration: `none`,
      '&:not(:last-of-type)': {
        borderBottom: 'var(--theme-ui-colors-panel-divider)'
      },
      '&:hover, &:focus': {
        backgroundColor: 'var(--theme-ui-colors-panel-background)'
      }
    }
  },

  radii: {
    default: '4px',
    card: '8px',
  },

  global: {
    '@keyframes wobble': {
      '0%, 100%': { transform: 'rotate(0deg)' },
      '15%': { transform: 'rotate(-15deg)' },
      '30%': { transform: 'rotate(10deg)' },
      '45%': { transform: 'rotate(-10deg)' },
      '60%': { transform: 'rotate(5deg)' },
      '75%': { transform: 'rotate(-5deg)' },
    },
    '@media (prefers-reduced-motion: reduce)': {
      '.emoji': {
        animation: 'none !important',
      },
    },
  },

  styles: {
    root: {
      color: `text`,
      display: `flex`,
      flexDirection: `column`,
      minHeight: `100vh`,

      fontFamily: 'body',
      fontWeight: 'body',
      lineHeight: 'body'
    },

    p: {
      fontSize: [2, 3]
    },

    ul: {
      fontSize: [2, 3]
    },

    outlined: {
      border: `4px solid #efefef`
    },

    table: {
      backgroundColor: 'light',
      color: 'tableText',
      width: '100%',
      borderCollapse: 'collapse',
      borderSpacing: 0,
      marginBottom: '1.5rem',
      overflow: 'hidden',
      borderRadius: '10px',
      boxShadow: '0 0 10px rgba(0, 0, 0, 0.1)',
      'th, td': {
        fontSize: '1.125em',
        textAlign: 'left',
        padding: '12px 15px',
        borderBottom: '1px solid',
        borderColor: 'muted',
      },
      'th': {
        backgroundColor: '#f4f4f9',
        fontWeight: 'bold',
        textTransform: 'uppercase',
        letterSpacing: '0.05em',
        borderTop: '2px solid',
        borderColor: 'muted',
      },
      'tbody tr:nth-of-type(odd)': {
        backgroundColor: '#fafafa',
      },
    },

    text: {
      inverse: {
        color: `muted`
      },
      title: {
        fontFamily: 'fonts.sans'
      }
    },

    GitHubCardFooter: {
      display: `flex`,
      justifyContent: `space-between`,
      mt: 2
    },

    PageFooter: {
      zIndex: '10',
      color: `text`,
      a: {
        color: `text`
      },
      width: `100%`,
      display: `block`
    },

    Book: {
      filter: theme => `drop-shadow(${theme.shadows.default})`,
      '&:hover, &:focus': {
        filter: theme => `drop-shadow(${theme.shadows.xl})`,
        transform: `scale(1.01)`,
        transition: `all .35s ease-in-out`
      }
    },

    Container: {
      py: [2, 3],
      px: [3, 4]
    },

    GradientBanner,

    GradientBannerDark: {
      ...GradientBanner,
      backgroundImage: `
        linear-gradient(#252e3c, #252e3c),
        linear-gradient(270deg, #00D7B9, #B95DD7 50%, #FFB367 100%);`,
      color: `light`
    },

    IntroExperienceSlide: {
      opacity: 0,
      height: 0,
      display: `flex`,
      visibility: `hidden`,
      '&.active-slide': {
        display: `block`,
        height: `auto`,
        opacity: 1,
        visibility: `initial`
      }
    },

    Header: {
      alignItems: `center`,
      color: `text`,
      display: `block`,
      transition: `all 0.3s ease-in-out`,
      width: `100%`
    },

    InstagramItem: {
      ...floatOnHover,
      background: `none`,
      border: `none`,
      boxShadow: `md`,
      cursor: `pointer`,
      overflow: `hidden`,
      borderRadius: `8px`,
      p: 0
    },

    TopNavigation: {
      color: `white`
    },

    VideoWrapper: {
      position: `relative`,
      paddingBottom: `56.25%` /* 16:9 */,
      paddingTop: `25px`,
      height: 0,
      iframe: {
        position: `absolute`,
        top: 0,
        left: 0,
        width: `100%`,
        height: `100%`
      }
    },

    WidgetFooter: {
      color: `text`,
      fontFamily: `heading`,
      textAlign: [`center`, `right`]
    },

    '.footnotes': {
      fontSize: theme => theme.fontSizes[1]
    },

    '.text-center': {
      textAlign: `center`
    },
  },

  text: {
    title: {
      color: `primary`,
      fontFamily: `heading`,
      fontSize: [1, 2],
      fontWeight: `550`,
      textTransform: `uppercase`
    }
  },

  variants: {
    cards: {
      dark: {
        backgroundColor: 'teal'
      }
    }
  }
})
