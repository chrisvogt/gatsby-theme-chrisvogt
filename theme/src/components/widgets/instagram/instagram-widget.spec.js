import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import { Provider as ReduxProvider } from 'react-redux'
import '@testing-library/jest-dom'
import configureStore from 'redux-mock-store'
import InstagramWidget from './instagram-widget'
import { ThemeUIProvider } from 'theme-ui'
import theme from '../../../gatsby-plugin-theme-ui'

jest.mock('../../../hooks/use-site-metadata', () =>
  jest.fn(() => ({
    instagramUsername: 'test_username',
    instagramDataSource: 'test_data_source'
  }))
)

jest.mock('lightgallery/react', () =>
  jest.fn(({ onInit }) => {
    onInit({ instance: mockLightGalleryInstance })
    return <div data-testid='lightgallery-mock' />
  })
)

jest.mock('lightgallery/plugins/thumbnail', () => jest.fn())
jest.mock('lightgallery/plugins/zoom', () => jest.fn())
jest.mock('lightgallery/plugins/video', () => jest.fn())
jest.mock('lightgallery/plugins/autoplay', () => jest.fn())

jest.mock('../../../actions/fetchDataSource', () =>
  jest.fn(() => ({
    type: 'FETCH_DATASOURCE'
  }))
)

const mockStore = configureStore([])
const mockLightGalleryInstance = {
  openGallery: jest.fn()
}

describe('InstagramWidget', () => {
  let store

  beforeEach(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {}) // Silence expected errors
    store = mockStore({
      widgets: {
        instagram: {
          state: 'SUCCESS',
          data: {
            collections: {
              media: [
                {
                  id: '123',
                  caption: 'Test Caption',
                  cdnMediaURL: 'https://cdn.chrisvogt.me/images/fake-instagram-image.jpg',
                  mediaType: 'IMAGE',
                  permalink: 'https://instagram.com/p/test'
                }
              ]
            },
            metrics: [
              { displayName: 'Followers', id: '1', value: 100 },
              { displayName: 'Following', id: '2', value: 50 }
            ]
          }
        }
      }
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
    console.error.mockRestore()
  })

  it('renders without crashing', () => {
    render(
      <ReduxProvider store={store}>
        <ThemeUIProvider theme={theme}>
          <InstagramWidget />
        </ThemeUIProvider>
      </ReduxProvider>
    )
    expect(screen.getByText(/Instagram/i)).toBeInTheDocument()
  })

  it('renders placeholders when isLoading is true', () => {
    const loadingStore = mockStore({
      widgets: {
        instagram: {
          state: 'LOADING',
          data: null
        }
      }
    })

    render(
      <ReduxProvider store={loadingStore}>
        <ThemeUIProvider theme={theme}>
          <InstagramWidget />
        </ThemeUIProvider>
      </ReduxProvider>
    )

    const placeholders = screen.getAllByText('', { selector: '.image-placeholder' })
    expect(placeholders.length).toBeGreaterThan(0)
  })

  it('renders media items when isLoading is false', () => {
    render(
      <ReduxProvider store={store}>
        <ThemeUIProvider theme={theme}>
          <InstagramWidget />
        </ThemeUIProvider>
      </ReduxProvider>
    )

    const thumbnails = screen.getAllByAltText(/Instagram post thumbnail/i)
    expect(thumbnails).toHaveLength(1)
    expect(thumbnails[0]).toHaveAttribute(
      'src',
      expect.stringContaining('https://cdn.chrisvogt.me/images/fake-instagram-image.jpg')
    )
  })

  it('toggles between "Show More" and "Show Less"', () => {
    render(
      <ReduxProvider store={store}>
        <ThemeUIProvider theme={theme}>
          <InstagramWidget />
        </ThemeUIProvider>
      </ReduxProvider>
    )

    const button = screen.getByText(/Show More/i)
    fireEvent.click(button)

    expect(screen.getByText(/Show Less/i)).toBeInTheDocument()

    fireEvent.click(screen.getByText(/Show Less/i))
    expect(screen.getByText(/Show More/i)).toBeInTheDocument()
  })

  it('opens LightGallery when handleClick is called', () => {
    render(
      <ReduxProvider store={store}>
        <ThemeUIProvider theme={theme}>
          <InstagramWidget />
        </ThemeUIProvider>
      </ReduxProvider>
    )

    const thumbnails = screen.getAllByAltText(/Instagram post thumbnail/i)
    fireEvent.click(thumbnails[0])

    expect(mockLightGalleryInstance.openGallery).toHaveBeenCalledWith(0)
  })

  it('assigns lightGalleryRef correctly on initialization', () => {
    const mockInstance = {}
    jest.mock('lightgallery/react', () =>
      jest.fn(({ onInit }) => {
        onInit({ instance: mockInstance })
        return <div data-testid='lightgallery-mock' />
      })
    )

    render(
      <ReduxProvider store={store}>
        <ThemeUIProvider theme={theme}>
          <InstagramWidget />
        </ThemeUIProvider>
      </ReduxProvider>
    )

    expect(mockInstance).toBeDefined()
  })

  it('updates focusedItem state on focus, blur, mouse enter, and mouse leave', () => {
    render(
      <ReduxProvider store={store}>
        <ThemeUIProvider theme={theme}>
          <InstagramWidget />
        </ThemeUIProvider>
      </ReduxProvider>
    )

    // Locate the <button> wrapping the media item
    const mediaItemButton = screen.getByRole('button', { name: /Instagram post thumbnail/i })
    expect(mediaItemButton).toBeInTheDocument() // Ensure the button is rendered

    // Focus event
    fireEvent.focus(mediaItemButton)
    expect(mediaItemButton.className).toContain('media-item--focused')

    // Blur event
    fireEvent.blur(mediaItemButton)
    expect(mediaItemButton.className).not.toContain('media-item--focused')

    // Mouse enter event
    fireEvent.mouseEnter(mediaItemButton)
    expect(mediaItemButton.className).toContain('media-item--focused')

    // Mouse leave event
    fireEvent.mouseLeave(mediaItemButton)
    expect(mediaItemButton.className).not.toContain('media-item--focused')
  })
})
