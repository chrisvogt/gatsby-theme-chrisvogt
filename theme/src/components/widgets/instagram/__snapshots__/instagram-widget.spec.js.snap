// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instagram Widget matches the loading state snapshot 1`] = `
<div
  className="css-12h1dx4-Widget"
  id="instagram"
>
  <header
    className="css-9v09o4-WidgetHeader"
  >
    <h2
      className="css-17p01r8"
    >
      Instagram
    </h2>
    <div
      className="css-1i3svai-WidgetHeader"
    >
      <svg
        className="svg-loaders-svg css-zmfjn6-CallToAction"
        fill="#1E90FF"
        height="24"
        viewBox="0 0 135 140"
        width="24"
      >
        <rect
          height={120}
          rx={6}
          width={15}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={30}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={140}
          rx={6}
          width={15}
          x={60}
        >
          <animate
            attributeName="height"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={90}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.25s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
        <rect
          height={120}
          rx={6}
          width={15}
          x={120}
          y={10}
        >
          <animate
            attributeName="height"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="120;110;100;90;80;70;60;50;40;140;120"
          />
          <animate
            attributeName="y"
            begin="0.5s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="10;15;20;25;30;35;40;45;50;0;10"
          />
        </rect>
      </svg>
    </div>
  </header>
  <div
    className="css-o0oe1r-ProfileMetricsBadge"
  >
    <div
      className="css-1stqq2l"
    >
       
    </div>
    <div
      className="css-zixz0n"
    >
       
    </div>
  </div>
  <div
    className="gallery"
  >
    <div
      className="css-1rcvf8r"
    >
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
      <div
        className="image-placeholder show-loading-animation"
      >
        <div
          className="rect-shape css-13iby3u"
          style={
            Object {
              "backgroundColor": "#efefef",
              "height": "100%",
              "marginRight": 10,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
